rholang/src/main/scala/coop/rchain/rholang/interpreter/SystemProcesses.scala

```diff
@@ -86,16 +86,23 @@ object SystemProcesses {
       private val UNLIMITED_MATCH_PHLO = matchListPar(Cost(Integer.MAX_VALUE))
 
       object ContractCall {
-        private def foldResult(
-            produceResult: Either[InterpreterError, Option[(ContWithMetaData, Channels)]]
-        ): F[Unit] =
-          produceResult.fold(
-            _ => F.raiseError(OutOfPhlogistonsError),
-            _.fold(F.unit) {
-              case (cont, channels) =>
-                dispatcher.dispatch(unpackCont(cont), channels.map(_.value), cont.sequenceNumber)
-            }
-          )
+        private def produce(rand : Blake2b512Random, sequenceNumber : Int)
+                           (values: Seq[Par], ch: Par): F[Unit] =
```

Comment:
Let's not glue this all together, as it's now unclear which parameter is used when. The second parameter list should be a part of an inner function.

I'd drop this commit and start again, applying the comments related to ContractCall object